corehq/apps/hqadmin/service_checks.py

```diff
@@ -222,6 +223,42 @@ def check_formplayer():
         return ServiceStatus(res.ok, msg)
 
 
+def check_pillowtop():
+    def changes_behind(pillow_meta):
+        lag = 0
+        for k, offset in pillow_meta['offsets'].items():
+            # seq_format is 'json' for KafkaChangeFeed and 'text' for
+            # ChangeFeed. The former seems to have offsets that are greater
+            # than the corresponding seq, while the latter has a seq value
+            # that's greater than the offset
+            if pillow_meta['seq_format'] == 'json':
+                seq = pillow_meta['seq'].get(k, 0)
+                lag += offset - seq
+            else:
+                seq = pillow_meta['seq']
+                lag += seq - offset
```

Comment:
unless i'm misunderstanding something I think this should also be `offset - seq`, and  if you were seeing `seq > offset` that's likely an artifact of the discrepancy mentioned by @snopoke below

Reply:
(also these names are super duper confusing. sorry if that's my fault - it probably is)

Reply:
When I checked this on prod, it looked like `seq` was consistently greater than `offset` when `seq_format == 'text'`, but less than `offset` when `seq_format == 'json'`.  I didn't pull samples at different times though, so I suppose it's possible that was just a coincidence.

Reply:
Yeah guessing was a coincidence. If you pull up [the system report](https://www.commcarehq.org/hq/admin/system/) at any given time you are bound to see a few couch feeds with offsets "ahead" of the checkpoints. Click the refresh button a few times for `UserCacheInvalidatePillow` to see this behavior. It has to do with each node having its own seq ID and therefore there not being a canonical sequence for the cluster as a whole. This metric is basically useless for cloudant.

Negative:
![image](https://user-images.githubusercontent.com/66555/40098350-5c6cfc3a-58db-11e8-96e9-74392afbe5d6.png)

Positive:

![image](https://user-images.githubusercontent.com/66555/40098372-71a08ff4-58db-11e8-9866-f1c2f70eed9c.png)

Reply:
Hmm, well that's unfortunate.  Thanks for the tip!